File: Code Generation/responses/python/deepseek/medium/apply-substitutions.py
from typing import List

class Solution:
    def applySubstitutions(self, replacements: List[List[str]], text: str) -> str:
        # Create a dictionary for quick lookup
        mapping = {key: value for key, value in replacements}
        
        # Define a recursive function to resolve placeholders
        def resolve(s: str) -> str:
            result = []
            i = 0
            n = len(s)
            while i < n:
                if i < n - 2 and s[i] == '%' and s[i + 2] == '%' and s[i + 1].isalpha():
                    # Found a placeholder %X%
                    key = s[i + 1]
                    placeholder_content = '%' + key + '%'
                    # Resolve the replacement value recursively
                    resolved_value = resolve(mapping[key])
                    result.append(resolved_value)
                    i += 3  # Skip the placeholder %X%
                else:
                    result.append(s[i])
                    i += 1
            return ''.join(result)
        
        # Resolve the entire text
        return resolve(text)

File: Code Generation/responses/cpp/deepseek/medium/apply-substitutions.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <functional>
#include <regex>

using namespace std;

class Solution {
public:
    string applySubstitutions(vector<vector<string>> replacements, string text) {
        unordered_map<string, string> dict;
        for (const auto& pair : replacements) {
            dict[pair[0]] = pair[1];
        }
        
        function<string(string)> expand = [&](string s) -> string {
            regex pattern("%([A-Z])%");
            smatch match;
            string result = s;
            
            while (regex_search(result, match, pattern)) {
                string key = match[1].str();
                string replacement = dict[key];
                result = regex_replace(result, regex("%" + key + "%"), expand(replacement));
            }
            
            return result;
        };
        
        return expand(text);
    }
};